packages/pyright-internal/src/analyzer/importResolver.ts

```diff
@@ -2429,44 +2397,52 @@ export class ImportResolver {
                         fileWithoutExtension,
                         leadingDots,
                         parentNameParts,
-                        sourceFilePath,
+                        sourceFileUri,
                         execEnv,
                         strictOnly
                     )
                 ) {
                     return;
                 }
 
-                suggestions.set(fileWithoutExtension, combinePaths(currentPath, file));
+                suggestions.set(fileWithoutExtension, file);
             }
         });
 
         entries.directories.forEach((dir) => {
-            if (filter && !dir.startsWith(filter)) {
+            const dirSuggestion = dir.basename;
+            if (filter && !dirSuggestion.startsWith(filter)) {
                 return;
             }
 
             if (
-                !this._isUniqueValidSuggestion(dir, suggestions) ||
-                !this._isResolvableSuggestion(dir, leadingDots, parentNameParts, sourceFilePath, execEnv, strictOnly)
+                !this._isUniqueValidSuggestion(dirSuggestion, suggestions) ||
+                !this._isResolvableSuggestion(
+                    dirSuggestion,
+                    leadingDots,
+                    parentNameParts,
+                    sourceFileUri,
+                    execEnv,
+                    strictOnly
+                )
             ) {
                 return;
             }
 
-            const initPyiPath = combinePaths(currentPath, dir, '__init__.pyi');
+            const initPyiPath = dir.initStubUri;
             if (this.fileExistsCached(initPyiPath)) {
-                suggestions.set(dir, initPyiPath);
+                suggestions.set(dirSuggestion, initPyiPath);
                 return;
             }
 
-            const initPyPath = combinePaths(currentPath, dir, '__init__.py');
+            const initPyPath = dir.initFileUri;
```

Comment:
Maybe `initPyUri` and `initPyiUri` would be more clear than `initFileUri` and `initStubUri`?

Reply:
Will change.